### PR TITLE
fixed the issue :  shared link viz data is getting overwritten

### PR DESCRIPTION
**Pull request by johibkhan2**

### CODE REVIEW

**Comment by admin-vizcentric** [comment]
There were no errors, go have a cup of coffee...

File: src/main/java/com/flair/bi/web/websocket/FbEngineWebSocketService.java [hunk 1]
```diff
@@ -39,6 +39,25 @@ public void pushGRPCMetaDeta(QueryResponse queryResponse, String request) {
         messagingTemplate.convertAndSendToUser(queryResponse.getUserId(), "/exchange/metaData", queryResponse.getData(), header);
     }
 
+    /**
+     * Send meta to users subscribed on channel "/user/exchange/metaData".
```
Comment (ir73) [inline src/main/java/com/flair/bi/web/websocket/FbEngineWebSocketService.java:43]: java doc seems to be wrong?

File: src/main/java/com/flair/bi/web/websocket/FbEngineWebSocketService.java [hunk 2]
```diff
@@ -39,6 +39,25 @@ public void pushGRPCMetaDeta(QueryResponse queryResponse, String request) {
...
+     * <p>
+     * The message will be sent only to the user with the given username.
+     *
+     * @param queryResponse query response
+     * @param request request
+     */
+    public void pushGRPCMetaDataSharedLink(QueryResponse queryResponse, String request) {
```
Comment (ir73) [inline src/main/java/com/flair/bi/web/websocket/FbEngineWebSocketService.java:50]: a lot of code duplication with `pushGRPCMetaDeta` method. Might make more sense to extract a common part?

Comment (ir73) [inline src/main/java/com/flair/bi/web/websocket/FbEngineWebSocketService.java:50]: method name does not seem to be describing the method body. There seem to be nothing related to shared link in this method, so perhaps method name could be improved?

File: src/main/java/com/flair/bi/service/GrpcQueryService.java [hunk 3]
```diff
@@ -219,6 +220,11 @@ private void callGrpcBiDirectionalAndPushInSocket(Datasource datasource, String
             public void onNext(QueryResponse queryResponse) {
                 log.debug("Finished trip with===" + queryResponse.toString());
                 fbEngineWebSocketService.pushGRPCMetaDeta(queryResponse, request);
+                if(request.equals(SHARED_LINK)){
+                    fbEngineWebSocketService.pushGRPCMetaDataSharedLink(queryResponse, request);
+                }else{
+                    fbEngineWebSocketService.pushGRPCMetaDeta(queryResponse, request);
```
Comment (ir73) [inline src/main/java/com/flair/bi/service/GrpcQueryService.java:226]: seems like `pushGRPCMetaDeta` is twice now - line 222 and line 226?

**Pull request merged**